src/python/pants/option/options_fingerprinter.py

```diff
@@ -121,6 +124,25 @@ def _assert_in_buildroot(self, filepath):
                          .format(filepath=filepath, buildroot=root))
       return filepath
 
+  def _fingerprint_dirs(self, dirpaths, topdown=True, onerror=None, followlinks=False):
+    """Returns a fingerprint of the given file directories and all their sub contents.
+
+    This assumes that the file directories are of reasonable size
+    to cause memory or performance issues.
+    """
+    hasher = sha1()
+    for dirpath in dirpaths:
+      for dirpath, dirnames, filenames in os.walk(dirpath, topdown=topdown, onerror=onerror,
```

Comment:
Because `os.walk` is likely to return things in different orderings on different machines, you'll need to sort paths here. Most likely what you'll want to do is collect all filenames under each of the top-level directories (where ordering does matter), and then sort them before calling `self._fingerprint_files(..)`.

But since you've iterated so much on this, I'm fine with doing it as a quick followup?

Reply:
nice catch, we should also sort the top-level dirpaths that is passed in as well.

Reply:
No, I don't think so. See the comment in `self._fingerprint_files` on that subject. The order of the entries in the list might matter.